nipype/interfaces/dtitk/registration.py

```diff
@@ -1,394 +1,431 @@
-                          argstr="-out %s", name_source="in_tensor",
-                          name_template="%s_diffeoxfmd.nii.gz")
-
-
-class diffeoSymTensor3DVolOutputSpec(TraitedSpec):
-    out_file = traits.File(desc='cheese', exists=True)
-
-
-class diffeoSymTensor3DVolTask(CommandLineDtitk):
-    """
-    Applies diffeomorphic transform to a tensor volume
-
-                Example
-                -------
-
-                >>> import nipype.interfaces.dtitk as dtitk
-                >>> node = dtitk.diffeoSymTensor3DVolTask()
-                >>> node.inputs.in_tensor = 'diffusion.nii'
-                >>> node.inputs.in_xfm = 'ants_Warp.nii.gz'
-                >>> node.run() # doctest: +SKIP
-                """
-
-    input_spec = diffeoSymTensor3DVolInputSpec
-    output_spec = diffeoSymTensor3DVolOutputSpec
-    _cmd = 'deformationSymTensor3DVolume'
-
-    def _list_outputs(self):
-        outputs = self.output_spec().get()
-        outputs['out_file'] = self.inputs.out_file
-        return outputs
-
 
 class affSymTensor3DVolInputSpec(CommandLineInputSpec):
-    in_tensor = traits.Str(desc='moving tensor', exists=True, mandatory=False,
-                           position=0, argstr="-in %s")
-    in_xfm = traits.Str(desc='transform to apply', exists=True,
-                        mandatory=False, position=1, argstr="-trans %s")
-    in_target = traits.Str(desc='', exists=True, mandatory=False, position=2,
-                           argstr="-target %s")
-    out_file = traits.Str(desc='', exists=True, mandatory=False, position=3,
-                          argstr="-out %s", name_source="in_tensor",
-                          name_template="%s_affxfmd.nii.gz")
+    in_file = File(desc='moving tensor volume', exists=True,
+                   argstr="-in %s", mandatory=True)
+    out_file = File(desc='output filename',
+                    argstr="-out %s", name_source="in_file",
+                    name_template="%s_affxfmd", keep_extension=True)
+    transform = File(exists=True, argstr="-trans %s",
+                     xor=['target', 'translation', 'euler', 'deformation'],
+                     desc='transform to apply: specify an input transformation\
+                      file; parameters input will be ignored',)
+    interpolation = traits.Enum('LEI', 'EI', usedefault=True,
+                                argstr="-interp %s",
+                                desc='Log Euclidean/Euclidean Interpolation')
+    reorient = traits.Enum('PPD', 'NO', 'FS', argstr='-reorient %s',
+                           usedefault=True, desc='Reorientation strategy: \
+                           preservation of principal direction, no \
+                           reorientation, or finite strain')
+    target = File(exists=True, argstr="-target %s", xor=['transform'],
+                  desc='output volume specification read from the target volume\
+                   if specified')
```

Comment:
For strings that go on beyond line lengths, the preferred way to do this would be:

```Python
    desc='output volume specification read from the target volume'
    ' if specified'
```

As long as you're within parentheses, newlines don't matter, and two strings sitting next to each other are automatically combined into a single string.

Reply:
Thanks. This doesn't work for the doctests, though (see, below)... is there a different syntax I should use?

Expected:

   'deformationScalarVolume -in im1.nii -interp 0 -out im1_diffeoxfmd.nii'
    ' -trans im_warp.df.nii'
Got:
    'deformationScalarVolume -in im1.nii -interp 0 -out im1_diffeoxfmd.nii -trans im_warp.df.nii'

Reply:
Ah, yeah, the doctests aren't as nice about these things. But they should be forgiving on whitespace. You may be able to throw in a newline without closing and reopening the quotes, but I'm not positive of that. If you go beyond the line limit in the doctests, I wouldn't quibble over it.